### PR TITLE
Availability blocker UI

### PR DESCRIPTION
**Pull request by mayhuberman**

### CODE REVIEW

File: ui/app/scripts/services/apolloApiService.js [hunk 1]
```diff
@@ -293,24 +293,26 @@ function ApiService($q, $http){
...
     };
 
-    var updateBlocker = function (id, name, environmentId, serviceId, stackId, isActive, blockerTypeName, blockerJsonConfiguration) {
+    var updateBlocker = function (id, name, environmentId, serviceId, stackId, availability, isActive, blockerTypeName, blockerJsonConfiguration) {
         return $http.put(CONFIG.appUrl + "blocker-definition/" + id, {
             name: name,
             environmentId: environmentId,
             serviceId: serviceId,
             stackId: stackId,
+            availability: availability,
```
Comment (ifatregev) [inline ui/app/scripts/services/apolloApiService.js]: Better naming?

File: ui/app/scripts/controllers/blockersCtrl.js [hunk 2]
```diff
@@ -87,6 +91,7 @@ angular.module('apollo')
           var serviceId = null;
           var environmentId = null;
           var stackId = null;
+          var availability = null;
```
Comment (ifatregev) [inline ui/app/scripts/controllers/blockersCtrl.js]: Make sure that null is acceptable in the backend side

File: ui/app/scripts/controllers/blockersCtrl.js [hunk 3]
```diff
@@ -104,6 +109,10 @@ angular.module('apollo')
               stackId = $scope.blockerStack.id;
           }
 
+          if ($scope.blockerAvailability) {
+              availability = $scope.blockerAvailability;
+          }
+
```
Comment (ifatregev) [inline ui/app/scripts/controllers/blockersCtrl.js]: $scope.blockerAvailability && availability = $scope.blockerAvailability

File: ui/app/scripts/controllers/blockersCtrl.js [hunk 4]
```diff
@@ -191,11 +202,14 @@ angular.module('apollo')
       function updateBlockers() {
           apolloApiService.getAllBlockers().then(function(response) {
               var tempBlockers = {};
+              var tempAvailabilities = {};
```
Comment (ifatregev) [inline ui/app/scripts/controllers/blockersCtrl.js]: better naming, order/list?

File: ui/app/scripts/controllers/blockersCtrl.js [hunk 5]
```diff
@@ -11,6 +11,9 @@ angular.module('apollo')
       $scope.timeBasedBlockerStartTime = {date: new Date()};
       $scope.timeBasedBlockerEndTime = {date: new Date()};
       $scope.allStacks = [];
+      $scope.allAvailabilities = [];
+      $scope.allBlockersAvailabilities = [];
```
Comment (ifatregev) [inline ui/app/scripts/controllers/blockersCtrl.js]: By naming we can't know what is the different purpose for ```allAvailabilities``` vs ```allBlockersAvailabilities```

File: ui/app/scripts/services/apolloApiService.js (hunk 1 above)
Comment (mayhuberman) [reply ui/app/scripts/services/apolloApiService.js]: This is the name of the field on the DB

File: ui/app/scripts/controllers/blockersCtrl.js (hunk 2 above)
Comment (mayhuberman) [reply ui/app/scripts/controllers/blockersCtrl.js]: Changed it directly to "var availability = $scope.blockerAvailability;" like the other String variables

**Pull request merged**